.github/workflows/test-install.yml

```diff
@@ -222,12 +222,27 @@ jobs:
         sudo apt update
         sudo apt install postgresql graphviz
 
-    - name: Upgrade pip and setuptools
-      # It is crucial to update `setuptools` or the installation of `pymatgen` can break
+    - name: Upgrade pip
       run: |
-        pip install --upgrade pip setuptools
+        pip install --upgrade pip
         pip --version
 
+    - name: Build pymatgen with compatible numpy
+      run: |
+        # This step is necessary because certain versions of `pymatgen` will not specify an explicit version of
+        # `numpy` in its build requirements, and so the latest version will be used. This causes problems,
+        # however, because this means that the compiled version of `pymatgen` can only be used with that version
+        # of `numpy` or higher, since `numpy` only guarantees forward compatibility of the ABI. If we want to
+        # run with an older version of `numpy`, we need to ensure that `pymatgen` is built with that same
+        # version. This we can accomplish by installing the desired version of `numpy` manually and then calling
+        # the install command for `pymatgen` with the `--no-build-isolation` flag. This flag will ensure that
+        # build dependencies are ignored and won't be installed (preventing the most recent version of `numpy`
+        # to be installed) and the build relies on those requirements already being present in the environment.
+        # We also need to install `wheel` because otherwise the `pymatgen` build will fail because `bdist_wheel`
+        # will not be available.
+        pip install numpy==1.21.4 wheel
+        pip install pymatgen==2022.0.16 --no-cache-dir --no-build-isolation
+
```

Comment:
Unless we also change our installation documentation, we **cannot** add this step here. The steps carried out as part of this workflow are particular designed to check that the **steps documented as part of the user documentation** for installing this package are working as expected. It seems like we will have to make that more obvious as part of the action's documentation.

Furthermore, afaik, the test-install flow has not failed due to the pymatgen issue, so there should be no need to "fix" it in this way either.

Reply:
Fair enough, will revert this. That being said, the existing workflow explicitly updates `pip` and `setuptools` adding a comment that the latter is critical for an install of `pymatgen`. This is not reflected in the [documentation](https://aiida.readthedocs.io/projects/aiida-core/en/latest/intro/install_system.html#intro-get-started-system-wide-install). But maybe this is no longer applicable and should also be removed?

Reply:
Indeed, maybe we add a comment to the docs that we test against the latest version of setuptools and pip and if there are any issues, users should try to update them? We could also explicitly test against the minimal version supported according to our [build settings](https://github.com/aiidateam/aiida-core/blob/c743a335480f8bb3a5e4ebd2463a31f9f3b9f9b2/pyproject.toml#L2) and the latest version?

My inclination would be to not make this more complicated until we run into problems.

Reply:
Seems reasonable to me. But I would suggest we leave this for another PR as it seems beyond the scope of getting the CI back up and running.